react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      imgIndex: 0,
+      activeInterval: null
+    }
+    this.changeIndex = this.changeIndex.bind(this)
+    this.slideShow = this.slideShow.bind(this)
+  }
+  slideShow() {
+    if (this.state.state.activeInterval) {
+      clearInterval(this.state.activeInterval)
+      this.setState({ activeInterval: null })
+    } else {
+      const newInterval = setInterval(this.changeIndex, 1000)
+      this.setState({ activeInterval: newInterval })
+    }
+  }
+  changeIndex(n) {
+    const slides = document.getElementsByClassName("slide");
+    this.setState({ imgIndex: n })
+
+    if (n > slides.length - 1) {
+      n = 0
+      this.setState({ imgIndex: n })
+    }
+    if (n < 0) {
+      n = slides.length - 1
+      this.setState({ imgIndex: n })
+    }
```

Comment:
This should be structured a little different by using else if statements so you don't have the extra setState call here on line 24.

```JSX
if (n >= slides.length) {
  ...
} else if (n < 0) {
  ...
} else {
  ...
}
```